integrations/firehydrant/client.py

```diff
@@ -0,0 +1,157 @@
+            logger.error(
+                f"HTTP error with status code: {e.response.status_code} and response text: {e.response.text}"
+            )
+            raise
+
+    async def get_paginated_resource(
+        self, resource_type: str
+    ) -> AsyncGenerator[list[dict[str, Any]], None]:
+        logger.info(f"Getting {resource_type} from Firehydrant")
+        endpoint = endpoint_resource_type_mapper.get(resource_type, resource_type)
+
+        pagination_params: dict[str, Any] = {
+            "page": 1,
+            "per_page": PAGE_SIZE,
+        }
+
+        try:
+            while True:
+                response = await self.send_api_request(
+                    endpoint=endpoint, query_params=pagination_params
+                )
+                if response.get("data"):
+                    yield response["data"]
+                else:
+                    logger.warning(f"No {resource_type} found in the response")
+
+                pagination_params["page"] += 1
+
+                if pagination_params["page"] > response.get("pagination", {}).get(
+                    "pages"
+                ):
+                    break
+
+        except Exception as e:
+            logger.error(f"Error while fetching {resource_type}: {str(e)}")
+
+    async def get_single_incident(self, incident_id: str) -> dict[str, Any]:
+        cache_key = f"{CacheKeys.INCIDENT}-{incident_id}"
+        if cache := event.attributes.get(cache_key):
+            return cache
+        incident_data = await self.send_api_request(endpoint=f"incidents/{incident_id}")
+        event.attributes[cache_key] = incident_data
+        return incident_data
+
+    async def get_single_service(self, service_id: str) -> dict[str, Any]:
+        serice_data = await self.send_api_request(endpoint=f"services/{service_id}")
+        service_analytics_data = await self.get_milestones_by_incident(
+            serice_data["active_incidents"]
+        )
+        serice_data["__incidents"] = service_analytics_data
+        return serice_data
+
+    async def get_single_environment(self, environment_id: str) -> dict[str, Any]:
+        return await self.send_api_request(endpoint=f"environments/{environment_id}")
+
+    async def get_single_retrospective(self, report_id: str) -> dict[str, Any]:
+        report_data = await self.send_api_request(
+            endpoint=f"post_mortems/reports/{report_id}"
+        )
+        incident_id = report_data["incident"]["id"]
+        tasks = await self.get_tasks_by_incident(incident_id=incident_id)
+        report_data["__incident"] = tasks
+        return report_data
+
+    async def get_tasks_by_incident(self, incident_id: str) -> dict[str, Any]:
+        logger.info(f"Getting tasks details for incident: {incident_id}")
+        task_endpoint = f"incidents/{incident_id}/tasks"
+        tasks = []
+
+        async for item in self.get_paginated_resource(task_endpoint):
+            tasks.extend(item)
+        return {"tasks": tasks}
```

Comment:
```suggestion
 return list(self.get_paginated_resource(task_endpoint))
```

Reply:
thanks @yairsimantov20 I tried to wrap the generator inside the list() to evaluate the whole result but it didn't work that's why I had to use the loop to process it one by one